geomstats/invariant_metric.py

```diff
@@ -57,9 +57,26 @@ def inner_product_at_identity(self, tangent_vec_a, tangent_vec_b):
             tangent_vec_a = gs.to_ndarray(tangent_vec_a, to_ndim=2)
             tangent_vec_b = gs.to_ndarray(tangent_vec_b, to_ndim=2)
 
+            n_tangent_vec_a = tangent_vec_a.shape[0]
+            n_tangent_vec_b = tangent_vec_b.shape[0]
+
+            assert (tangent_vec_a.shape == tangent_vec_a.shape
+                    or n_tangent_vec_a == 1
+                    or n_tangent_vec_b == 1)
```

Comment:
Does this assert work in TF without the eager execution mode?

Reply:
Yes, I ran the tf script [here](https://github.com/geomstats/geomstats/issues/155#issue-371004876) and set a breakpoint at the location. It passes this location and provide correct assert values:

```
/vol/medic01/users/bh1511/_venv/geomstats-py3/bin/python3.6 /Applications/PyCharm.app/Contents/helpers/pydev/pydevd.py --cmd-line --multiproc --qt-support=auto --client 127.0.0.1 --port 50116 --file /vol/medic01/users/bh1511/_build/geomstats-farrell/tests/bh_test_tf.py
pydev debugger: process 11617 is connecting
Connected to pydev debugger (build 182.4505.26)
/vol/medic01/users/bh1511/_venv/geomstats-py3/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: compiletime version 3.5 of module 'tensorflow.python.framework.fast_tensor_util' does not match runtime version 3.6
  return f(*args, **kwds)
Using tensorflow backend
import sys; print('Python %s on %s' % (sys.version, sys.platform))
Python 3.6.3 |Intel Corporation| (default, May  3 2018, 23:25:54) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.0.1 -- An enhanced Interactive Python. Type '?' for help.
PyDev console: using IPython 7.0.1
Python 3.6.3 |Intel Corporation| (default, May  3 2018, 23:25:54) 
[GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.31)] on darwin
In[2]: tangent_vec_a.shape == tangent_vec_b.shape
Out[2]: True
In[3]: n_tangent_vec_a
Out[3]: Dimension(4)
In[4]: n_tangent_vec_b
Out[4]: Dimension(4)

```